support-lib/jni/djinni_support.cpp

```diff
@@ -239,8 +239,13 @@ jint JniEnum::ordinal(JNIEnv * env, jobject obj) const {
 
 LocalRef<jobject> JniEnum::create(JNIEnv * env, jint value) const {
     LocalRef<jobject> values(env, env->CallStaticObjectMethod(m_clazz.get(), m_staticmethValues));
+    jniExceptionCheck(env);
     DJINNI_ASSERT(values, env);
-    return LocalRef<jobject>(env, env->GetObjectArrayElement(static_cast<jobjectArray>(values.get()), value));
+    LocalRef<jobject> result(env,
+                             env->GetObjectArrayElement(static_cast<jobjectArray>(values.get()),
+                                                        value));
+    jniExceptionCheck(env);
+    return std::move(result);
```

Comment:
Yet the opposite case is fine, where the storage location of the return value is declared const, but used as non-const inside the function?  I'm not sure I understand why the non-constness of the return value would matter.  I guess I've thought of const as simply a limitation on behavior.  What are the cases where the compiler would generate something which is actually unsafe so as to need to be UB?  I don't know of any cases where an object's representation or behavior is different because it was constructed const, vs. being constructed non-const, and then accessed through a const reference.  But I'm not a compiler optimization expert, I suppose.